File: loan-frontend/src/services/loanAccountService.js
import api from "../api";

export const getAllLoanAccounts = async () => {
  const response = await api.get("/api/loan-accounts");
  return response.data;
};

export const createLoanAccount = async (loanApplicationId) => {
  const response = await api.post("/api/loan-accounts", {loanApplicationId,});
  return response.data;
};

export const getLoanAccountByAccountNumber = async (accountNumber) => {
  const response = await api.get(`/api/loan-accounts/account-number/${accountNumber}`);
  return response.data;
};

export const getLoanAccountByApplication = async (applicationId) => {
  const response = await api.get(`/api/loan-accounts/my/${applicationId}`);
  return response.data;
};

export const getLoanAccount = async (id) => {
  const response = await api.get(`/api/loan-accounts/${id}`);
  return response.data;
};

File: loan-backend/src/main/java/com/loan/backend/service/LoanAccountService.java
package com.loan.backend.service;

import com.loan.backend.dto.CreateLoanAccountRequest;
import com.loan.backend.entity.LoanAccount;

import java.util.List;

public interface LoanAccountService {

    LoanAccount createLoanAccount(
            CreateLoanAccountRequest request);
    List<LoanAccount> getAllLoanAccounts();
    LoanAccount getLoanAccount(Long id);
    LoanAccount getLoanAccountByApplicationId(
            Long applicationId);
    LoanAccount getLoanAccountByAccountNumber(
            String accountNumber);
}